src/createBaseForm.js

```diff
@@ -45,9 +45,9 @@ function createBaseForm(option = {}, mixins = []) {
         };
       },
 
-      componentWillReceiveProps(nextProps) {
+      componentWillReceiveProps(nextProps, nextContext) {
         if (mapPropsToFields) {
-          this.fields = mapPropsToFields(nextProps);
+          this.fields = mapPropsToFields(nextProps, nextContext);
```

Comment:
https://github.com/react-component/form/pull/69#discussion_r112400778 this is not a solution, it is confusing.